kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/tokenstore/KafkaTokenStore.java

```diff
@@ -0,0 +1,516 @@
+         *
+         * @param consumerConfiguration a timeout specifying the time after which this process will force a claim
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder consumerConfiguration(Map<String, Object> consumerConfiguration) {
+            assertMinimalValidClientConfiguration(consumerConfiguration,
+                                                  "The consumer configuration may not be null, and needs to contain a 'bootstrap-servers' value");
+            this.consumerConfiguration = setConsumerConfig(consumerConfiguration);
+            return this;
+        }
+
+        /**
+         * Sets the {@code producerConfiguration} specifying configuration for reading token updates.
+         *
+         * @param producerConfiguration a timeout specifying the time after which this process will force a claim
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder producerConfiguration(Map<String, Object> producerConfiguration) {
+            assertMinimalValidClientConfiguration(producerConfiguration,
+                                                  "The consumer configuration may not be null, and needs to contain a 'bootstrap-servers' value");
+            this.producerConfiguration = setProducerConfig(producerConfiguration);
+            return this;
+        }
+
+        /**
+         * Initializes a {@link KafkaTokenStore} as specified through this Builder.
+         *
+         * @return a {@link KafkaTokenStore} as specified through this Builder
+         */
+        public KafkaTokenStore build() {
+            return new KafkaTokenStore(this);
+        }
+
+        /**
+         * Validates whether the fields contained in this Builder are set accordingly.
+         *
+         * @throws AxonConfigurationException if one field is asserted to be incorrect according to the Builder's
+         *                                    specifications
+         */
+        protected void validate() throws AxonConfigurationException {
+            assertNotNullOrEmpty(topic, "The topic is a hard requirement and should be provided");
+            assertNonNull(serializer, "The Serializer is a hard requirement and should be provided");
+            assertNotNullOrEmpty(nodeId, "The nodeId is a hard requirement and should be provided");
+            assertMinimalValidClientConfiguration(consumerConfiguration,
+                                                  "Consumer configuration is a hard requirement and should at least contain a 'bootstrap-servers' value");
+            assertMinimalValidClientConfiguration(producerConfiguration,
+                                                  "Producer configuration is a hard requirement and should at least contain a 'bootstrap-servers' value");
+        }
+
+        private Map<String, Object> setConsumerConfig(Map<String, Object> configuration) {
```

Comment:
Curious: would these be the same/similar consumer and producer configs as for the event publication and reading in this extension?

Reply:
Kinda, idea would be they can be reused, with the specific changes for the topic, but at least security etc. is the same, and we could autowire them.